### PR TITLE
Suggest

### PR DESCRIPTION
**Pull request by lonegunmanb**
## Describe your changes

This pull request amended `.gitignore` file, updated github action files, fix some pr-check issues.

## Issue number

#000

## Checklist before requesting a review
- [ ] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [ ] I have executed pre-commit on my machine
- [ ] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

**Comment by lonegunmanb** [comment]
@matt-FFFFFF Would you please give this pr a review? Thanks!

**Comment by matt-FFFFFF** [comment]
> ## As this variable is a map[string]VirtualNetwork, do we need vnet's name here? Would using the key as vnet's name be better?

@lonegunmanb This is true, though the map key must be known at plan time. There are cases where a caller will want to have the virtual network names sourced from something that is only known after apply. Hence the decision to have the map key separate.

File: variables.tf [hunk 1]
```diff
@@ -1,24 +1,31 @@
 variable "hub_virtual_networks" {
   description = "A map of hub virtual networks to create"
   type = map(object({
+    ## As this variable is a map[string]VirtualNetwork, do we need vnet's name here? Would using the key as vnet's name be better?
```
Comment (matt-FFFFFF) [inline variables.tf:4]: Due constraint that map key is known at plan, there maybe a scenario where a caller uses other resource data to generate the vnet names, there I propose keeping vnet name distinct from map key.

File: variables.tf [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
     location            = string
     resource_group_name = string
 
-    bgp_community                   = optional(string, "")
-    ddos_protection_plan            = optional(string, "")
-    dns_servers                     = optional(list(string), [])
+    bgp_community = optional(string)
+    ddos_protection_plan = optional(object({
+      id     = string
+      enable = bool
```
Comment (matt-FFFFFF) [inline variables.tf:13]: For DDoS - I think we should simplify this for the caller, even through the API has these properties they aren't really necessary. I believe we should use an optional string and if it is supplied, then build the correct inputs to the submodule.

**Review by matt-FFFFFF — COMMENTED** [review]
Great work @lonegunmanb - is this now functional?

**Comment by lonegunmanb** [comment]
Hi @matt-FFFFFF I've improved this pr as you suggested, and I've verified that we can turn ddos plan off by removing this nested block, so I just simplify it as you suggested.

**Review by matt-FFFFFF — APPROVED** [review]
Lgtm!

**Pull request merged**